indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx

```diff
@@ -549,3 +618,173 @@ EditableListDisplay.propTypes = {
+        return a.selectedOption.localeCompare(b.selectedOption);
+      }
+      return textCmp;
+    });
+
+  const handleSearchChange = search => onChange({search, filters: activeFilters.filters});
+  const handleFiltersChange = filters => onChange({search: activeFilters.search, filters});
+
+  const addFilter = (filter, selectedOption) => {
+    const newFilter = {
+      key: `${filter.key}_${selectedOption}`,
+      filter,
+      selectedOption,
+    };
+    handleFiltersChange(sortActiveFilters([...activeFilters.filters, newFilter]));
+  };
+
+  const removeFilter = key => {
+    handleFiltersChange(activeFilters.filters.filter(f => f.key !== key));
+  };
+
+  const makeIsSelectedOption = filterKey => option =>
+    activeFilters.filters.some(f => f.key === `${filterKey}_${option}`);
+
+  return (
+    <div>
+      <Dropdown
+        text={Translate.string('Filter')}
+        icon="filter"
+        labeled
+        button
+        className={activeFilters.filters.length > 0 ? 'orange icon' : 'icon'}
+        onClose={() => setOpenSubmenu(-1)}
+      >
+        <Dropdown.Menu>
+          {activeFilters.filters.length > 0 ? (
+            <>
+              <div key="_filters" style={{display: 'flex', flexWrap: 'wrap'}}>
+                {activeFilters.filters.map(({key, filter, selectedOption}) => (
+                  <Label
+                    key={key}
+                    style={{marginTop: '0.4em', marginLeft: '0.4em', marginRight: '0.4em'}}
+                    className="fluid"
+                    color="orange"
+                  >
+                    {filter.text}
+                    <Icon
+                      name="delete"
+                      className="float right"
+                      onClick={evt => {
+                        evt.stopPropagation();
+                        setOpenSubmenu(-1);
+                        removeFilter(key);
+                      }}
+                    />
+                    <Label.Detail className="float right">
+                      {filter.options.find(o => o.value === selectedOption).text}
+                    </Label.Detail>
+                  </Label>
+                ))}
+              </div>
+              <Dropdown.Item
+                key="_clear"
+                text={Translate.string('Clear all filters')}
+                onClick={() => handleFiltersChange([])}
+              />
+            </>
+          ) : (
+            <Dropdown.Item text={Translate.string('No filters were added yet')} disabled />
+          )}
+          <Dropdown.Divider />
+          {_.sortBy(filterOptions, 'text').map(filter => (
+            <EditableListFilterSubmenu
+              key={filter.key}
+              filter={filter}
+              addFilter={addFilter}
+              open={openSubmenu === filter.key}
+              onOpen={() => setOpenSubmenu(filter.key)}
+              isSelectedOption={makeIsSelectedOption(filter.key)}
+            />
+          ))}
```

Comment:
What about simplifying it like this? `EditableListFilterSubmenu` would instead have `options` and `label` as props. 

```suggestion
          {_.sortBy(filterOptions, 'text').map(({key, options, text}) => (
            <EditableListFilterSubmenu
              key={key}
              options={options}
              label={text}
              addFilter={addFilter}
              open={openSubmenu === key}
              onOpen={() => setOpenSubmenu(key)}
              isSelectedOption={makeIsSelectedOption(key)}
            />
          ))}
```

Reply:
I'm wondering whether the `EditableListFilterSubmenu` is even necessary anymore, or should I just embed the `Dropdown` there directly